src/expression.js

```diff
@@ -547,6 +559,32 @@ function caml_binaryen_ref_i31(wasm_mod, typ) {
   return wasm_mod.ref.i31(typ);
 }
 
+//Provides: caml_binaryen_ref_test
+function caml_binaryen_ref_test(wasm_mod, ref, typ) {
+  return wasm_mod.ref.test(ref, typ);
+}
+
+//Provides: caml_binaryen_ref_cast
+function caml_binaryen_ref_cast(wasm_mod, ref, typ) {
+  return wasm_mod.ref.cast(ref, typ);
+}
+
+//Provides: caml_binaryen_br_on
+//Requires: caml_jsstring_of_string
+//Requires: Binaryen
+function caml_binaryen_br_on(wasm_mod, op, name, ref, typ) {
+  switch (op) {
+    case Binaryen.BrOnNull:
+      return wasm_mod.br_on.null(caml_jsstring_of_string(name), ref, typ);
+    case Binaryen.BrOnNonNull:
+      return wasm_mod.br_on.non_null(caml_jsstring_of_string(name), ref, typ);
+    case Binaryen.BrOnCast:
+      return wasm_mod.br_on.cast(caml_jsstring_of_string(name), ref, typ);
```

Comment:
Shouldn't this be `br_on_cast`
https://github.com/WebAssembly/binaryen/blob/6ec7b5f9c615d3b224c67ae221d6812c8f8e1a96/src/js/binaryen.js-post.js#L2488